cli/cli.go

```diff
@@ -366,43 +347,54 @@ func runCliCmd(cmd *cobra.Command, args []string, flags *CliCmdFlags, clientImag
 	}
 
 	if flags.run {
-		if err = runAndShowContainers(*flags.services, flags); err != nil {
-			return []error{err}
+		if utils.Contains(*flags.services, "validator") {
+			*flags.services = append(*flags.services, "validator-import")
 		}
-	} else {
-		// Let the user decide to see the instructions for executing the scripts and exit or let the tool execute them
-		if err = runScriptOrExit(flags); err != nil {
+		if err := buildContainers(*flags.services, flags.generationPath); err != nil {
 			return []error{err}
 		}
-	}
-
-	if !flags.noValidator {
-		log.Info(configs.ValidatorTips)
-
-		// Run validator after execution and consensus clients are synced, unless the user intencionally wants to run the validator service in the previous step
-		if !utils.Contains(*flags.services, validator) {
-			// Wait for clients to start
-			// log.Info(configs.WaitingForNodesToStart)
-			// time.Sleep(waitingTime)
-			// Track sync of execution and consensus clients
-			// TODO: Parameterize wait arg of trackSync
-			if err = trackSync(monitor, results.ELPort, results.CLPort, time.Minute*5, flags); err != nil {
+		if flags.slashingProtection != "" {
+			// Setup wait for validator import
+			exitCh, errCh := serviceManager.Wait(services.ServiceCtValidatorImport, dockerct.WaitConditionNextExit)
+			// Run validator-import
+			if err := runAndShowContainers([]string{"validator-import"}, flags); err != nil {
```

Comment:
Currently, we are always running the `validator-import` service before the validator if the user wants to run a validator client, and importing the slashing data no make sense without a validator, we need to run the `validator-import` to set up the validator database and be able to import the data. I think the solution can be to check that the user wants to run a validator client when using the `slashing-import` flag, otherwise, its usage is wrong.

Reply:
In order to import slashing data you need the validator datadir, to set up the validator datadir you need the `validator-import` and to run the `validator-import` you need a keystore. Is true we are always running `validator-import` but this is wrong, we should run only the `validator-import` service if there is a keystore.
```mermaid
flowchart LR
subgraph flow
A(import-slashing)-- needs --> B(validator datadir)
C(validator-import) -- gets --> D(keystore) -- prepare --> B
end
subgraph result
AA(import-slashing) -- depends on --> CC(validator-import) -- depends on --> DD(keystore)
end
flow ==> result
```
This code will change as the `cli` command will be highly interactive. So not worry about if the user wants to run a validator or not. I'm more focused on the slashing import code, as this will also be a separate non-interactive command and in this case we need to make sure there is a provided keystore. 

Checking for the keystore will be very useful as we will be able to warn the user that there is no keystore in the `cli` command.